Mail/Mailer.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using System.IO;
+using System.Text;
+using MailKit.Net.Smtp;
+using MimeKit;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace NetCoreBootstrap.Mail
+{
+    public class Mailer
+    {
+        private readonly SmtpClient _smtpClient;
+        private string _host, _username, _password, _name, _email, _jsonFilePath;
+        private int _hostPort;
+        private char _jsonSplitter = ':';
+        
+        public Mailer()
+        {
+            _smtpClient = new SmtpClient();
+            _jsonFilePath = "appsettings.Development.json";
+            SetAccountConfiguration();
+        }
+
+        public void Send(string toName, string toAddress, string subject, string body, string type = "plain")
+        {
+            var message = new MimeMessage { Subject = subject, Body = new TextPart (type) { Text = body } };
+            message.From.Add(new MailboxAddress(Name, Email));
+			message.To.Add(new MailboxAddress(toName, toAddress));
+
+            using(var client = Client)
+            {
+                try
+                {
+                    client.Connect(Host, HostPort);
+                    client.AuthenticationMechanisms.Remove("XOAUTH2");
+                    if(String.IsNullOrEmpty(Username) || String.IsNullOrEmpty(Password)) throw new ArgumentNullException();
+                    client.Authenticate(Username, Password);
+                    client.Send(message);
+                    client.Disconnect(true);
+                }
+                catch(Exception e)
+                {
+                    throw new Exception(message: e.Message);
+                }
+            }
+        }
+
+        private void SetAccountConfiguration()
+        {
+            try
+            {
+                var resourceFileStream = new FileStream(_jsonFilePath, FileMode.Open, FileAccess.Read, FileShare.Read, 4096, 
+                                                        FileOptions.Asynchronous | FileOptions.SequentialScan);
+                using (resourceFileStream)
+                {
+                    var resourceReader = new JsonTextReader(new StreamReader(resourceFileStream, Encoding.UTF8, detectEncodingFromByteOrderMarks: true));
+                    using (resourceReader)
+                    {
+                        var resource = JObject.Load(resourceReader);
+                        if(resource == null) throw new ArgumentNullException();
+                        _host = TryGetValue(resource, "Mailer:Host").ToString();
```

Comment:
if you need set this property out of constructor use the set method or don´t use like readonly ("_")

Reply:
and apply same for the next properties :P